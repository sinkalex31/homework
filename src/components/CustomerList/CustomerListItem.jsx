import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
    Button,
    ListGroupItem,
    ListGroupItemHeading,
    ListGroupItemText,
    Modal,
    ModalHeader,
    ModalBody,
    ModalFooter,
} from 'reactstrap';

import { CustomerPropTypes } from 'utils/propTypes';

export default class CustomerListItem extends Component {
    static propTypes = {
        customer: CustomerPropTypes.isRequired,
    };

    state = {
        toggle: false,
    };

    toggle = () => this.setState({
        modal: !this.state.modal,
    });

    render () {
        const { customer: { id, email, metadata: { fullName } }, removeCustomer } = this.props;

        return (
            <ListGroupItem
                action
            >
                <Link to={`/customer/${id}`}>
                    <ListGroupItemHeading>{fullName}</ListGroupItemHeading>
                    <ListGroupItemText>{email}</ListGroupItemText>
                </Link>
                <Modal isOpen={this.state.modal} toggle={this.toggle}>
                    <ModalHeader toggle={this.toggle}>
                        Remove customer {fullName}?
                    </ModalHeader>
                    <ModalBody>
                        Are you sure you want to remove this customer information?
                    </ModalBody>
                    <ModalFooter>
                        <Button color="primary" onClick={() => {
                            removeCustomer(id);
                            this.toggle();
                        }}>
                            Yes
                        </Button>
                        <Button color="secondary" onClick={this.toggle}>Cancel</Button>
                    </ModalFooter>
                </Modal>
                <Button onClick={this.toggle}>Remove</Button>
            </ListGroupItem>
        );
    }
};