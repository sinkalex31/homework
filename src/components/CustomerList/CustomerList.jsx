import React from 'react';
import { ListGroup } from 'reactstrap';
import PropTypes from 'prop-types';

import { CustomerPropTypes } from 'utils/propTypes';
import CustomerListItem from './CustomerListItem';

const CustomerList = ({ customers, removeCustomer }) => (
    <ListGroup>
        {customers.map((customer) => (
            <CustomerListItem
                key={customer.id}
                customer={customer}
                removeCustomer={removeCustomer}
            />
        ))}
    </ListGroup>
);

CustomerList.propTypes = {
    customers: PropTypes.arrayOf(CustomerPropTypes).isRequired,
    removeCustomer: PropTypes.func.isRequired,
};

export default CustomerList;