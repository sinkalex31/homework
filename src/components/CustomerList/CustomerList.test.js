import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import CustomerList from './CustomerList';

const customers = [
    {
        id: '1',
        metadata: {
            date: new Date('2018.01.10 14:00'),
        },
    },
    {
        id: '2',
        metadata: {
            date: new Date('2018.02.15 10:00'),
        },
    },
];

describe('CustomerList', () => {
    it ('should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <CustomerList customers={customers} removeCustomer={() => {}}/>
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});