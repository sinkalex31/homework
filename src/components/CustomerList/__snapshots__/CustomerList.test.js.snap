// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerList should render correctly 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item-action list-group-item"
  >
    <a
      href="/customer/1"
      onClick={[Function]}
    >
      <h5
        className="list-group-item-heading"
      />
      <p
        className="list-group-item-text"
      />
    </a>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </li>
  <li
    className="list-group-item-action list-group-item"
  >
    <a
      href="/customer/2"
      onClick={[Function]}
    >
      <h5
        className="list-group-item-heading"
      />
      <p
        className="list-group-item-text"
      />
    </a>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </li>
</ul>
`;
